packages/protocol/contracts/governance/Proposals.sol

```diff
@@ -0,0 +1,348 @@
+   *   votes is returned.
+   * @param proposal The proposal struct.
+   * @param quorum The minimum participation at which "no" votes are not added.
+   * @return The support ratio with the quorum condition.
+   */
+  function getSupportWithQuorumPadding(
+    Proposal storage proposal,
+    FixidityLib.Fraction memory quorum
+  )
+    internal
+    view
+    returns (FixidityLib.Fraction memory)
+  {
+    uint256 yesVotes = proposal.votes.yes;
+    if (yesVotes == 0) {
+      return FixidityLib.newFixed(0);
+    }
+    uint256 noVotes = proposal.votes.no;
+    uint256 totalVotes = yesVotes.add(noVotes).add(proposal.votes.abstain);
+    uint256 requiredVotes = quorum.multiply(
+        FixidityLib.newFixed(proposal.networkWeight)
+    ).fromFixed();
+    if (requiredVotes > totalVotes) {
+      noVotes = noVotes.add(requiredVotes.sub(totalVotes));
+    }
+    return FixidityLib.newFixedFraction(yesVotes, yesVotes.add(noVotes));
+  }
+
+  /**
+   * @notice Returns the stage of a dequeued proposal.
+   * @param proposal The proposal struct.
+   * @param stageDurations The durations of the dequeued proposal stages.
+   * @return The stage of the dequeued proposal.
+   * @dev Must be called on a dequeued proposal.
+   */
+  function getDequeuedStage(
+    Proposal storage proposal,
+    StageDurations storage stageDurations
+  )
+    internal
+    view
+    returns (Stage)
+  {
+    uint256 stageStartTime = proposal.timestamp.add(stageDurations.approval).add(
+      stageDurations.referendum
+    ).add(stageDurations.execution);
+    // solhint-disable-next-line not-rely-on-time
+    if (now >= stageStartTime) {
+      return Stage.Expiration;
+    }
+    stageStartTime = stageStartTime.sub(stageDurations.execution);
+    // solhint-disable-next-line not-rely-on-time
+    if (now >= stageStartTime) {
+      return Stage.Execution;
+    }
+    stageStartTime = stageStartTime.sub(stageDurations.referendum);
+    // solhint-disable-next-line not-rely-on-time
+    if (now >= stageStartTime) {
+      return Stage.Referendum;
+    }
+    return Stage.Approval;
+  }
+
+  /**
+   * @notice Returns the number of votes cast on the proposal over the total number
+   *   of votes in the network as a fraction.
+   * @param proposal The proposal struct.
+   * @return The participation of the proposal.
+   */
+  function getParticipation(
+    Proposal storage proposal
+  ) 
+    internal 
+    view 
+    returns (FixidityLib.Fraction memory) 
+  {
+    uint256 totalVotes = proposal.votes.yes.add(proposal.votes.no).add(proposal.votes.abstain);
+    return FixidityLib.newFixedFraction(totalVotes, proposal.networkWeight);
+  }
+
+  /**
+   * @notice Returns a specified transaction in a proposal.
+   * @param proposal The proposal struct.
+   * @param index The index of the specified transaction in the proposal's transaction list.
+   * @return The specified transaction.
+   */
+  function getTransaction(
```

Comment:
Where is this being used? Would it make sense for it instead to return a `Transaction` struct, without unpacking?

Reply:
external view in Governance, not ever operated on